core/src/main/scala/kafka/cluster/Broker.scala

```diff
@@ -88,7 +90,7 @@ object Broker {
                 (ep.protocolType, ep)
               }.toMap
             }
-          val rack: Option[String] = brokerInfo.get("rack").asInstanceOf[Option[String]]
+          val rack = brokerInfo.get("rack").filter(_ != null).map(_.asInstanceOf[String])
```

Comment:
Interesting. I have to admit that I am still novice to Scala. Your code will return None and my code returns Some(null). I agree your solution is correct because at one place, I used this filter:

`    val brokersWithRack = brokers.filter(_.rack.nonEmpty)
`

and Some(null) does not return the desired result.

Thanks for catching this.